packages/help-center/src/components/help-center-gpt.tsx

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { LoadingPlaceholder } from '@automattic/components';
+import { HelpCenterSelect, useJetpackSearchAIQuery } from '@automattic/data-stores';
+import styled from '@emotion/styled';
+import { useSelect } from '@wordpress/data';
+import { external, Icon, page } from '@wordpress/icons';
+import { useI18n } from '@wordpress/react-i18n';
+import { useEffect } from 'react';
+import stripTags from 'striptags';
+import './help-center-article-content.scss';
+import useTyper from '../hooks/use-typer';
+import { HELP_CENTER_STORE } from '../stores';
+
+const GPTResponsePlaceholder = styled( LoadingPlaceholder )< { width?: string } >`
+	:not( :last-child ) {
+		margin-block-end: 0.5rem;
+	}
+	width: ${ ( { width = '100%' } ) => width };
+`;
+
+const GPTResponseDisclaimer = styled.div`
+	color: var( --studio-gray-50 );
+	font-size: 11px;
+	text-align: right;
+
+	@media ( min-width: 600px ) {
+		padding: 5px 0;
+	}
+`;
+
+interface Props {
+	loadingMessage: string;
+}
+
+const LoadingPlaceholders: React.FC< Props > = ( { loadingMessage } ) => (
+	<>
+		<p className="help-center-gpt-response__loading">{ loadingMessage }</p>
+		<GPTResponsePlaceholder width="80%" />
+		<GPTResponsePlaceholder width="80%" />
+		<GPTResponsePlaceholder width="55%" />
+	</>
+);
+
+export function HelpCenterGPT() {
+	const { __ } = useI18n();
+
+	// Report loading state up.
+	const { message } = useSelect(
+		( select ) => ( {
+			message: ( select( HELP_CENTER_STORE ) as HelpCenterSelect ).getMessage(),
+		} ),
+		[]
+	);
+
+	const query = message ?? '';
+
+	// First fetch the links
+	const { data: links, isError: isLinksError } = useJetpackSearchAIQuery(
+		'9619154',
+		query,
+		'urls'
+	);
+
+	// Then fetch the response
+	const { data, isError: isResponseError } = useJetpackSearchAIQuery(
+		'9619154',
+		links?.urls ? query : '',
+		'response'
+	);
+
+	const allowedTags = [ 'a', 'p', 'ol', 'ul', 'li', 'br', 'b', 'strong', 'i', 'em' ];
+
+	const isGPTError = isLinksError || isResponseError;
+
+	useEffect( () => {
+		if ( data?.response ) {
+			recordTracksEvent( 'calypso_helpcenter_show_gpt_response', {
+				location: 'help-center',
+			} );
+		}
+	}, [ data ] );
+
+	const loadingMessages: string[] = [
+		__( 'Finding relevant support documentation…', __i18n_text_domain__ ),
+		__( 'Gathering all the data…', __i18n_text_domain__ ),
+		__( 'Thanks for your patience, this might take a bit…', __i18n_text_domain__ ),
+		__( 'Processing the information found…', __i18n_text_domain__ ),
+		__( "I'm still writing, thank you for your patience!", __i18n_text_domain__ ),
+		__( 'Any minute now…', __i18n_text_domain__ ),
+		__( 'Really, any minute now…', __i18n_text_domain__ ),
+	];
+
+	const loadingMessage = useTyper( loadingMessages, true, {
+		delayBetweenCharacters: 80,
+		delayBetweenWords: 1000,
+	} );
```

Comment:
It would be nice to do `loadingMessage || '&nbsp;'` or change the hook to return nbsp instead of nothing. Currently, the UI is jumping.

https://github.com/Automattic/wp-calypso/assets/17054134/7b830dd5-6bad-4559-97a0-86025145e963